File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    l = 0
    r = len(people) - 1
    people.sort()
    while l <= r:
        if people[l] + people[r] <= limit:
            answer += 1
            l += 1
            r -= 1
        else:
            answer += 1
            r -= 1
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    sort(people.begin(), people.end());
    int left = 0;
    int right = people.size()-1;
    
    while(left <= right){
        if(people[left]+people[right] <= limit){
            left+=1;
            right-=1;
            answer+=1;
        }
        else{
            answer+=1;
            right-=1;
        }
    }
    return answer;
}

//50 50 70 80